prow/pubsub/subscriber/subscriber.go

```diff
@@ -58,6 +61,81 @@ type prowCfgClient interface {
 	GetPostsubmitsStatic(identifier string) []config.Postsubmit
 }
 
+type InRepoConfigCacheGetter struct {
+	CookieFilePath string
+	CacheSize      int
+	Agent          *config.Agent
+	mu             sync.RWMutex
+	GithubOptions  flagutil.GitHubOptions
+	DryRun         bool
+
+	CacheMap map[string]*config.InRepoConfigCache
+}
+
+func (irc *InRepoConfigCacheGetter) readCacheMap(key string) (*config.InRepoConfigCache, bool) {
+	irc.mu.RLock()
+	defer irc.mu.RUnlock()
+
+	res, ok := irc.CacheMap[key]
+	return res, ok
+}
+
+func (irc *InRepoConfigCacheGetter) getCache(cloneURI, host string) (*config.InRepoConfigCache, error) {
+	if irc.CacheMap == nil {
+		irc.CacheMap = map[string]*config.InRepoConfigCache{}
+	}
+
+	var key string
+	// We are using github
+	if irc.GithubOptions.TokenPath != "" || irc.GithubOptions.AppPrivateKeyPath != "" {
+		key = irc.GithubOptions.Host
+	} else {
+		key = cloneURI
+	}
+
+	if cache, ok := irc.readCacheMap(key); ok {
```

Comment:
not sure you need a helper here, I think you can do irc.mu.RLock as the first thing into the function, and unlock right after here, unlocking without defer is totally fine

Reply:
Done